### PR TITLE
Authorization & User object

### PR DESCRIPTION
**Pull request by seiferla**

### CODE REVIEW

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java:1]: You can change this to a record

`public record User`

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return this.password.equals(password);
+  }
+
+  public boolean checkEmail(String email) {
+    return this.email.equals(email);
+  }
+
+  public boolean checkUser(String email, String password) {
+    return checkEmail(email) && checkPassword(password);
+  }
```
[the remark this answers is not in the document]
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java:32]: So basically you can decide if you want to use the InMemory or the File solution. Originally, the File solution was planned and has a better persistence.

[the remark this answers is not in the document]
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java:32]: For the File solution, you can serialize and deserialize a Set of Users with a new GSON instance.

**Comment by seiferla** [comment]
I'm almost finished. I still have to write some tests for the FileUserRepo and improve the SonarLint. But you can already review

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/parsing/MailboxParsingService.java [hunk 2]
```diff
@@ -18,21 +19,29 @@ public class MailboxParsingService {
...
+    this.mailboxFileProvider = mailboxRepository;
+    this.mailboxParser = mailboxParser;
+    this.authorizationService = authorizationService;
+  }
+
+  public Mailbox getMailboxOfAddress(User user) throws MailboxParsingServiceException {
+
+    if (!authorizationService.authorize(user.email(), user.password())) {
+      throw new MailboxParsingServiceException("User not authorized");
     }
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/parsing/MailboxParsingService.java:37]: It is not a Exception of the ParsingService. It is an Exception of the Authorization Service. You could just rewrite it like this:

```suggestion
  public Mailbox getMailboxOfAddress(User user) throws MailboxParsingServiceException, AuthorizationException{
      authorizationService.authorize(user);
  }
```

And in the authorizationService it should throw an AuthorizationException("User not authorized");

File: src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/testUsers.json
Comment (Jonas-Karl) [inline src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/testUsers.json:1]: this can be removed

**Review by Jonas-Karl — COMMENTED** [review]
Please move the user Exceptions to a seperate folder inside the user domain.

Also you should add tests for the AuthenticationService

File: src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/UserTest.java
Comment (Jonas-Karl) [inline src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/UserTest.java:1]: This tests recreates a users.json on every run on the filesystem. Please have a look how I handled a similar issue with a @TempFile here: https://github.com/DHBW-Mailflow/mailserver/pull/91/files#diff-94f39b5d33268ad4015d82da6cb26e370e9afa5984ebd0e22d5f3929716de72f

**Review by Jonas-Karl — APPROVED** [review]
Done in pair programming

**Pull request merged**